src/htcp.cc

```diff
@@ -923,7 +923,8 @@ htcpSpecifier::checkHit()
         return;
     }
 
-    if (!checkHitRequest->header.parse(req_hdrs, reqHdrsSz)) {
+    Http::ContentLengthInterpreter interpreter;
+    if (!checkHitRequest->header.parse(req_hdrs, reqHdrsSz, interpreter)) {
```

Comment:
I would add an HttpRequest::parseHeader(buffer, size) method that can be used here and in similar contexts. The method will declare ContentLengthInterpreter, of course. It is wrong to reach into an object to manipulate its parts that may depend on other object fields (e.g., request method) and adding this HttpRequest method would be a step in the right direction even without the need to hide ContentLengthInterpreter.